std/fs/expand_glob.ts

```diff
@@ -196,12 +205,15 @@ export function* expandGlobSync(
     } else if (globSegment == "..") {
       const parentPath = joinGlobs([walkInfo.filename, ".."], globOptions);
       try {
-        return yield* [
-          { filename: parentPath, info: statSync(parentPath) }
-        ].filter(shouldInclude);
-      } catch {
-        return;
+        if (shouldInclude(parentPath)) {
+          return yield { filename: parentPath, info: statSync(parentPath) };
+        }
+      } catch (error) {
+        if (error.kind != ErrorKind.NotFound) {
+          throw error;
+        }
```

Comment:
It seems this little 3 line bit of code is repeated many times. Should it be abstracted into a function?